### PR TITLE
KV: add add/update functions, unit tests

### PR DESCRIPTION
**Pull request by bradjc**
Improvements to the KV support to match the kernel.

### CODE REVIEW

File: examples/tests/kv/main.c [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+    "kvtestappah",
+    "kvtestappav",
+    "kvtestappa7",
+  };
+
+  int ret;
+
+  for (uint32_t i = start; i < stop; i++) {
+    // printf ("%d: %s\n", i, array[i]);
+
```
Comment (ppannuto) [inline examples/tests/kv/main.c:285]:
```suggestion
```

**Review by ppannuto — APPROVED** [review]

**Review by ppannuto — DISMISSED** [review]

File: examples/tests/kv_interactive/main.c [hunk 2]
```diff
@@ -122,6 +124,60 @@ int main(void) {
...
+      printf("Adding %s=%s\n", key_buf, value_buf);
+
+      ret = kv_system_add_sync(key_buf, key_len, value_buf, value_len);
+      if (ret < 0) {
+        printf("Error adding key %i\n", ret);
+      } else {
+        printf("Added key-value\n");
+      }
+
+    } else if (strncmp(read_buf, "update", 3) == 0) {
```
Comment (alistair23) [inline examples/tests/kv_interactive/main.c:154]: Shouldn't this be a 7?

Comment (bradjc) [reply examples/tests/kv_interactive/main.c]: Fixed.

**Comment by bradjc** [comment]
Merge this?

**Review by ppannuto — APPROVED** [review]

**Pull request merged**